Remove readonly keyword from buffer pointer definitions

aidanfnv opened the pull request:
For #6880

This change removes the `readonly` keyword from buffer pointer definitions from the GLSL source emitter, to allow for mutable buffer pointers.
Support for `readonly` will be readded when we add `const` pointer support later.

csyonghe reviewed (commented):
Let's not try to be clever here and just remove `readonly` for all cases.

We will support `readonly` when we add `const` pointer support later. For now we should assume all pointers are not mutable.

aidanfnv commented:
Understood. I have removed all of what I added and instead just removed readonly from the emitted GLSL, and updated the PR description and title accordingly.

csyonghe commented:
Need to update the expected test result for the existing const-buffer-pointer test.

jkwak-work commented:
Does this PR resolve the problem with playground?
